Samples/MediaPlayerCS/MainPage.xaml.cs

```diff
@@ -85,6 +108,270 @@ public MainPage()
             CoreWindow.GetForCurrentThread().KeyDown += MainPage_KeyDown;
 
             StreamDelays.AddHandler(Slider.PointerReleasedEvent, new PointerEventHandler(StreamDelayManipulation), true);
+
+            subtitleDispatcherTimer.Interval = TimeSpan.FromMilliseconds(33);
+            subtitleDispatcherTimer.Tick += RenderSubtitleDispatcherTimer_Tick;
+
+            mediaPlayer.PlaybackSession.PlaybackStateChanged += PlaybackSession_PlaybackStateChanged;
+        }
+
+        private async void PlaybackSession_PlaybackStateChanged(MediaPlaybackSession sender, object args)
+        {
+            await Dispatcher.RunAsync(CoreDispatcherPriority.Normal, async () =>
```

Comment:
Making COM and WinRT events async is always a bit dangerous because - in this case - it causes a closure on the MediaPlaybackSession object for the lambda.
Depending on the upstream implementation this may cause issues or not. In case of WebView2 for example, doing so causes crashes. Sometimes even when executing lengthy non-async code.

Anyway, there's no need for an await here.

Reply:
This is normal async code. I don't see how this causes problem, except if there is special/buggy behavior on WebView2. Sure it does capture the sender in lambda, as long as the async method runs. Once it is finished, capture is released and instance can be free'd. This is normal, and nothing to worry about as far as i see.

Reply:
It's unlikely the playback session will be released, as it is a singleton inside MediaPlayer and there's a circular reference between them.

Reply:
> This is normal async code. I don't see how this causes problem, except if there is special/buggy behavior on WebView2.

It has to do with COM event invocation and I've seen this in quite a number of cases over the decades. 
When you encounter such case at some time you might remember what I said today, possibly even solve an issue that others failed to resolve. :-)